### DISCUSSION TITLE
'Error: Structure option :noinline unrecognized' when installing

### DISCUSSION THREAD

**Discussion by jax2net** [joaotavora/eglot#754] [Q&A] [answered]
When installing eglot from both ELPA/MELPA and from the source file, I get the following error.
`eglot.el:69:1:Error: Structure option :noinline unrecognized`
Inside eglot.el line 69 is a reference to `(require 'xref)`

I've also tried installing in an `emacs -q` session but the same error persists.

Any help is appreciated.

**Comment by joaotavora** [comment]
Thanks.

This sounds like a bug somwhere, but to make sure, we would need to provide this info in bug reporting tmplate.

Maybe we can do that here in this discussion.  Can you detail exactly what you mean by "I've also tried installing in an `emacs -q` session"?  That's the part that's more interesting, to me. In other words can you fill in the missing parts?

```
1. sth
2. sth
3. emacs -q
4. sth
5. sth
6. error happens
```

Also please say the version of emacs (what `emacs -v` reports) and your operating system.

Thanks in advance,

**Comment by jax2net** [comment, marked as answer]
On Linux Mint, using Emacs 26.3 installed from the package manager:
```
1. Install eglot
2. Receive above error
2. Try with emacs -q
3. Initialize MELPA
4. Install eglot 
5. Receive above error
```
I built Emacs 27.1 and got it working. I believe the error had to do with my specific Emacs install, as even `M-x lsp` gave the same error.

**Reply by joaotavora** [reply]
Ok thanks.  

Sigh.  I wonder at my utter failure in communicating simple ideas sometimes.

It's the "Try with `emacs -q`" that I wanted you to expand on :-D  Wasn't it clear that was looking for a way to experience the problem, i.e. to _reproduce_ it?  There's a lot of wiggle room in "try with emacs -q" and "Install eglot".

I got to find a better way to explain this.

Anyway, since you've resolved this, I'm also marking this discussion resolved, OK? Thanks

**Reply by jax2net** [reply]
Apologies, I realize my explanation was vague but these are the exact steps I took to reproduce the problem. I can try to answer any specific questions if you'd like.

Initially I thought it had to do with my init file but even after adding MELPA to a session without it I had the same problem, so I assumed it had to do with my specific install. Unfortunately I didn't save the contents of the debugger backtrace.

**Reply by joaotavora** [reply]
No problem, @jax2net .  I'm not saying these steps aren't "exact", but they're too coarsely described.

If you'd have said _"it doesn't work"_ that would also be "exact", but not very useful :-)  I'm trying to figure out a way to explain to bug reporters that they help me reproduce the problem from scratch, as if I myself were a terminal so to speak.  So I was hoping for something similar to:

```
1. git clone https://github.com/joaotavora/eglot
2. cd eglot
3. emacs -v   # to display the version
4. emacs -Q -L . -l eglot.el
5. C-x C-f go to a file that shows the error
6. M-x eglot
7. BOOM
```

**Reply by joaotavora** [reply]
Maybe you can help me.  In your opinion, what words should I have used to communicate to someone like you that this is what I was looking after?

**Reply by jax2net** [reply]
For me your words were fine; perhaps if I'd seen `bug_report.md` beforehand it would've been clearer, but I think this is an issue with me not putting in the requisite effort of reading issue related files and/or filing an actual bug report (instead of in the discussion tab).

Often when I get errors related to something simple (like package installation) the last thing I want to do is read more documentation, but in this case it probably would've made your life easier. I think in this situation it would've been best for me to fill out an actual bug report, though this would be annoying for me. Regardless I'll try and do my own due diligence in future bug reports. Thanks for your work on this package :)

**Reply by dgutov** [reply]
Hmm, looks like I've made the latest `xref` incompatible with Emacs 26: the `:noinline` option in `cl-defstruct` became available only starting with Emacs 27.

Seems like something I should really look into. Patches welcome, though.

**Reply by dgutov** [reply]
This is hopefully fixed in `xref` version 1.3.1.

**Reply by joaotavora** [reply]
Thanks!
